sns/parity_math.c

```diff
@@ -1835,22 +1841,45 @@ static void bufvec_buf(struct m0_bufvec *bvec, struct m0_buf *buf,
 	uint8_t		       *buf_data;
 	uint32_t		seg_size;
 
-	M0_ENTRY("bvec=%p, buf=%p, count=%u, dir=%s",
-		 bvec, buf, count, (dir?"TRUE":"FALSE"));
+	M0_ENTRY("bvec=%p, buf=%p, count=%u", bvec, buf, count);
 
 	for (j = 0, i = 0; j < count; ++j, i = 0) {
 		m0_bufvec_cursor_init(&cursor, &bvec[j]);
 		buf_data = (uint8_t *)buf[j].b_addr;
 		do {
 			seg_size = bvec[j].ov_vec.v_count[i];
-			if (dir)
-				memcpy(&buf_data[i * seg_size],
-				       m0_bufvec_cursor_addr(&cursor),
-				       seg_size);
-			else
-				memcpy(m0_bufvec_cursor_addr(&cursor),
-				       &buf_data[i * seg_size],
-				       seg_size);
+			memcpy(&buf_data[i * seg_size],
```

Comment:
Here, we assume the seg_size is all equal?

Reply:
Updated the function.